pkg/syncer/metadatasyncer.go

```diff
@@ -546,7 +546,7 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 							log.Infof("CSI full sync failed with error: %+v", err)
 						}
 					} else {
-						vcconfigs, err := cnsvsphere.GetVirtualCenterConfigs(ctx, configInfo.Cfg)
+						vcconfigs, err := cnsvsphere.GetVirtualCenterConfigs(ctx, metadataSyncer.configInfo.Cfg)
```

Comment:
do we need this change? configInfo points to metadataSyncer.configInfo assigned in InitMetadataSyncer()

Reply:
This is to make sure we have the latest configInfo. Suppose the config changes, then we update metadataSyncer.configInfo. This change will only get reflected in the fullSync if we use metadataSyncer.configInfo and not configInfo that we obtained in InitMetadataSyncer()